src/Data/Aeson/Internal/Functions.hs

```diff
@@ -11,22 +11,31 @@
 module Data.Aeson.Internal.Functions
     (
       mapHashKeyVal
+    , mapTextKeyVal
     , mapKeyVal
     , mapKey
     ) where
 
 import Prelude.Compat
 
 import Data.Hashable (Hashable)
+import qualified Data.Aeson.TextMap as TM
 import qualified Data.HashMap.Strict as H
 import qualified Data.Map as M
+import qualified Data.Text as T
 
 -- | Transform a 'M.Map' into a 'H.HashMap' while transforming the keys.
 mapHashKeyVal :: (Eq k2, Hashable k2) => (k1 -> k2) -> (v1 -> v2)
               -> M.Map k1 v1 -> H.HashMap k2 v2
 mapHashKeyVal fk kv = M.foldrWithKey (\k v -> H.insert (fk k) (kv v)) H.empty
 {-# INLINE mapHashKeyVal #-}
 
+-- | Transform a 'M.Map' into a 'H.HashMap' while transforming the keys.
```

Comment:
Probably missing something but figured it can't hurt to ask. Should this maybe be:

```suggestion
-- | Transform a 'M.Map' into a 'TM.TextMap' while transforming the keys.
```

?

Reply:
Thanks!